File: src/p0016/python/solution.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        ans = math.inf
        for i in range(len(nums)):
            j, k = i + 1, len(nums) - 1
            while j < k:
                s = nums[i] + nums[j] + nums[k]
                if s == target:
                    return target
                if abs(s-target) < abs(ans-target):
                    ans = s
                if s > target: k -= 1
                if s < target: j += 1
                
        return ans

File: src/p0016/golang/solution.go
func threeSumClosest(nums []int, target int) int {
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] < nums[j]
	})
	ans, minDis := 0, math.MaxInt
	for k := 0; k < len(nums); k++ {
		for i, j := k+1, len(nums)-1; i < j; {
			x := nums[i] + nums[j] + nums[k]
			if x == target {
				return x
			} else if x > target {
				if x-target < minDis {
					minDis = x - target
					ans = x
				}
				j--
			} else {
				if target-x < minDis {
					minDis = target - x
					ans = x
				}
				i++
			}
		}
	}
	return ans
}